### DISCUSSION TITLE
hcxdumptool -o (option) error!

### DISCUSSION THREAD

**Discussion by asentroy** [ZerBea/hcxdumptool#423] [General]
I tried to a single BSSID PMKID search then show me this error!
need solution!!

hcxdumptool -o testresult.pcapng -i wlan0mon --filterlist=f2-text.txt --filtermode=2 --enable_status=1 
hcxdumptool: invalid option -- 'o'
hcxdumptool 6.3.1 (C) 2023 by ZeroBeat
usage: hcxdumptool -h for help

**Comment by vertexgamer** [comment]
does it work without the filters? I know hcxdumptool uses a bpf file now with 6.3.2
I'm not sure if these options are still supported

**Comment by ZerBea** [comment]
Since c 6.3.0 options have changed due to performance reasons.
That has been mentioned in the changelog, the help menu (-h and --help) and the FAQ here in discussions:

```
hcxdumptool -o testresult.pcapng -i wlan0mon --filterlist=f2-text.txt --filtermode=2 --enable_status=1
-o repalced by -->  -w
--filterlist=f2-text.txt --> rpaced by --bpf
--filtermode=2 --> replaced by --bpf
--enable_status=1
BTW:
-i wlan0mon --> as mentioned in help it is not recommended to run hcxdumptool on a virtual monitor interface!!!!!! Do not use scripts like airmon-ng to set monitor mode before you run hcxdumptool, because hcxdumtool set  its own monitor mode!!!!!
```

The solutions:
read -h as suggested by hcxdumptool:
`usage: hcxdumptool -h for help`

```
read changelog:
03.05.2023
==========
hcxdumptool: changed/added several options:
$ hcxdumptool --help
```

read the FAQ:
https://github.com/ZerBea/hcxdumptool/discussions/330
https://github.com/ZerBea/hcxdumptool/discussions/343

**Comment by ZerBea** [comment]
@vertexgamer Yes it works without attaching a BPF. In that case hcxdumptool responds to every request and requests everything.
https://github.com/ZerBea/hcxdumptool/discussions/420
